### PR TITLE
Refactor controllers

### PR DESCRIPTION
**Pull request by MikelSage**
Reduces some logic in controllers, adds more flash messages.

### CODE REVIEW

**Comment by MikelSage** [comment]
@mikeyduece Come check this out bro.

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -5,11 +5,9 @@ def create
...
     @cart.add_item(item.id)
     session[:cart] = @cart.contents
-    if params[:to_cart]
-      redirect_to cart_path
-    else
-      redirect_to root_path
-    end
+    flash[:success] = "Added #{item.title} to your cart"
+
+    redirect_to request.referrer
```
Comment (jimszalew) [inline app/controllers/carts_controller.rb:10]: image

**Review by jimszalew — APPROVED** [review]
This is an amazing refactor. Very elegant.

**Comment by jimszalew** [comment]
This is really great work. Glad you found the request.referrer method

**Pull request merged**